### PR TITLE
Convert basic components to TS

### PR DESCRIPTION
**Pull request by ranquild**
Epic: https://github.com/metabase/metabase/issues/18345

Needed to unblock migrating homepage to TS

### CODE REVIEW

File: frontend/src/metabase/components/Link.tsx [hunk 1]
```diff
@@ -1,22 +1,27 @@
...
-import styled from "styled-components";
-import { display, color, hover, space } from "styled-system";
-
-import { stripLayoutProps } from "metabase/lib/utils";
 import Tooltip from "metabase/components/Tooltip";
+import { stripLayoutProps } from "metabase/lib/utils";
+import React, { ReactNode } from "react";
+import { Link as ReactRouterLink, LinkProps } from "react-router";
+import styled from "styled-components";
+import { color, display, hover, space } from "styled-system";
```
Comment (kulyk) [inline frontend/src/metabase/components/Link.tsx:7]: We usually keep all third-party imports at the top (see the [JS style guide note](https://github.com/metabase/metabase/blob/master/docs/developers-guide/frontend.md#javascript-conventions))

**Review by kulyk — COMMENTED** [review]
Looks great! Just a question on imports order here

File: frontend/src/metabase/components/Link.tsx (hunk 1 above)
Comment (ranquild) [reply frontend/src/metabase/components/Link.tsx]: Thanks, fixed!

**Review by kulyk — APPROVED** [review]
🎉

**Pull request merged**